packages/q/qtbase/xmake.lua

```diff
@@ -0,0 +1,228 @@
+        elseif package:is_plat("macosx") then
+            arch = "clang_64"
+        elseif package:is_plat("android") then
+            if version:le("5.13") or version:ge("6.0") then
+                if package:is_arch("x86_64", "x64") then
+                    arch = "android_x86_64"
+                elseif package:is_arch("arm64", "arm64-v8a") then
+                    arch = "android_arm64_v8a"
+                elseif package:is_arch("armv7", "armv7-a", "armeabi", "armeabi-v7a") then
+                    arch = "android_armv7"
+                elseif package:is_arch("x86") then
+                    arch = "android_x86"
+                else
+                    raise("unhandled arch " .. package:targetarch())
+                end
+            else
+                arch = "android"
+            end
+        end
+
+        local installdir = package:installdir()
+        print("host", host)
+        print("target", target)
+        print("versionstr", versionstr)
+        print("arch", arch)
+        os.vrunv("aqt", {"install-qt", "-O", installdir, host, target, versionstr, arch})
+
+        -- move files to root
+        os.mv(path.join(installdir, versionstr, "*", "*"), installdir)
+        os.rmdir(path.join(installdir, versionstr))
+
+        -- special case for cross-compilation since we need binaries we can run on the host
+        if (package:is_plat("mingw") and not is_host("windows")) then
+            local runhost
+            if is_host("linux") then
+                runhost = "linux"
+            elseif is_host("macosx") then
+                runhost = "mac"
+            else
+                raise("unhandled host " .. os.host())
+            end
+
+            -- download qtbase to bin_host folder
+            os.vrunv("aqt", {"install-qt", "-O", path.join(installdir, "bin_host"), runhost, "desktop", versionstr, "--archives", "qtbase"})
+
+            -- add symbolic links for useful tools
+            local tool_folders = {}
+            if version:ge("6.0") then
+                tool_folders.bin = {
+                    qmake = true,
+                    qmake6 = true
+                }
+
+                tool_folders.libexec = {
+                    moc = true,
+                    rcc = true,
+                    uic = true
+                }
+            else
+                tool_folders.bin = {
+                    qmake = true,
+                    moc = true,
+                    rcc = true,
+                    uic = true
+                }
+            end
+
+            for folder, tools in pairs(tool_folders) do
+                for _, file in pairs(os.files(path.join(installdir, "bin_host", versionstr, "*", folder, "*"))) do
+                    local filename = path.filename(file)
+                    if (tools[filename]) then
+                        local targetpath = path.join(installdir, folder, filename)
+                        os.rm(targetpath)
+                        os.ln(file, targetpath)

```

Comment:
please do not use softlink, it maybe does not work on windws, you can use `os.cp`

Reply:
this code should not be executed on windows, but I added a os.cp fallback in case it does in the future.